packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -37,6 +40,28 @@ class FloatingActionButton extends StatefulComponent {
 }
 
 class _FloatingActionButtonState extends State<FloatingActionButton> {
+  Performance _childSegue = new Performance(duration: _kFloatingActionButtonChildSegue);
+  Widget _child;
+
+  void initState() {
+    super.initState();
+    _child = new RotationTransition(
+      performance: _childSegue,
+      turns: new AnimatedValue<double>(-0.125, end: 0.0, curve: _kFloatingActionButtonChildSegueInterval),
+      child: config.child
+    );
+    _childSegue.play();
+  }
+
+  void didUpdateConfig(FloatingActionButton oldConfig) {
+    super.didUpdateConfig(oldConfig);
+    if (config.child.key == oldConfig.child.key && config.backgroundColor == oldConfig.backgroundColor)
+      return;
+    _childSegue
+      ..progress = 0.0
+      ..play();
+  }
```

Comment:
Don't you need to update the child in the RotationTransition?  Otherwise the child from the new config will not be used in the build.